OneCommunityGlobal/HighestGoodNetworkApp
# Yubo-fix_adding_professional_links

# Description

<img width="797" alt="Frontend_PR#1101_Description-1" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/a759643b-eea9-404c-a720-db36cffae5a5">

Fixes **User Management Component** # 21 **Urgent**

## Related PRS (if any):
This frontend PR is related to frontend PR #980:

1. It adds fixed **Google Doc** link and **Dropbox** link.

## Main changes explained:
- Rewrite the `EditLinkModal.jsx` file as the previous one has unfunctional and unnecessary `dispatch` functions.
- Add the style to 'help context' to tell user why certain links can't be added.
- Integrate PR #980.


## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as admin user
4. go to 'Other links'→ choose any user→ Edit their links below their Profile Picture.
5. verify if all the valid links can be added and removed. `Google Doc` and 'Dropbox link' are special **admin links** that need to be fixed, and invalid URLs can't be added
6. remember to click `Save Changes` after the `Modal` is closed.

## Screenshots or videos of changes:
<img width="992" alt="Frontend_PR#1101_Description-2" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/0751e380-0db2-4adf-910f-ed72d106837f">


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/e9c7ccc5-8ae3-41ee-922a-d1aed97efc78



## Note:
You may find if it can be saved by only clicking `update` button, it may be in a new PR once it's solved. Thank you for reviewing.
